hummingbot/connector/exchange/ftx/ftx_exchange.pyx

```diff
@@ -0,0 +1,898 @@
+                                                      order_type,
+                                                      decimal_price)
+            elif order_type is OrderType.MARKET:
+                order_result = await self.place_order(order_id,
+                                                      trading_pair,
+                                                      decimal_amount,
+                                                      True,
+                                                      order_type,
+                                                      None)
+
+            else:
+                raise ValueError(f"Invalid OrderType {order_type}. Aborting.")
+
+            exchange_order_id = str(order_result["result"]["id"])
+
+            tracked_order = self._in_flight_orders.get(order_id)
+            if tracked_order is not None and exchange_order_id:
+                tracked_order.update_exchange_order_id(exchange_order_id)
+                order_type_str = "MARKET" if order_type == OrderType.MARKET else "LIMIT"
+                self.logger().info(f"Created {order_type_str} buy order {order_id} for "
+                                   f"{decimal_amount} {trading_pair}")
+                self.c_trigger_event(self.MARKET_BUY_ORDER_CREATED_EVENT_TAG,
+                                     BuyOrderCreatedEvent(
+                                         self._current_timestamp,
+                                         order_type,
+                                         trading_pair,
+                                         decimal_amount,
+                                         decimal_price,
+                                         order_id
+                                     ))
+
+        except asyncio.CancelledError:
+            raise
+        except Exception:
+            tracked_order = self._in_flight_orders.get(order_id)
+            tracked_order.set_status("FAILURE")
+            self.c_stop_tracking_order(order_id)
+            self.logger().error(
+                f"Error submitting buy {order_type} order to ftx for "
+                f"{decimal_amount} {trading_pair} "
+                f"{decimal_price}. Response={order_result}",
+                exc_info=True
+            )
+            self.c_trigger_event(self.MARKET_ORDER_FAILURE_EVENT_TAG,
+                                 MarketOrderFailureEvent(
+                                     self._current_timestamp,
+                                     order_id,
+                                     order_type
+                                 ))
+
+    cdef str c_buy(self,
+                   str trading_pair,
+                   object amount,
+                   object order_type=OrderType.LIMIT,
+                   object price=NaN,
+                   dict kwargs={}):
+        cdef:
+            int64_t tracking_nonce = <int64_t> get_tracking_nonce()
+            str order_id = str(f"buy-{trading_pair}-{tracking_nonce}")
```

Comment:
Need order_id prefix or headers to identify hummingbot orders in FTX

Reply:
Should the order id be something like "FTX-buy-ETH-USD-<nonce>" instead? Most of the connectors seem to follow the model above.